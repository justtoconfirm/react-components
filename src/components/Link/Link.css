.link {
	color: inherit;
}

.link:hover {
	text-decoration: none;
}