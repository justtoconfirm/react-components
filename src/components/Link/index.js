import React from 'react';
import './Link.css';

const Link = ({ href, target = '', children }) => {

	// Check if the component has the target prop with the value of _blank to render as an external link
	const isExternalLink = target && (target.includes('_blank'));

	const renderAsExternalLink = () =>
		<a {...{href, target}} rel="noopener noreferrer" className="link">{children}</a>  

	const renderAsLink = () =>
		<a href={href} className="link">{children}</a> 
	
	return (

		// If the link component has a target prop with a value with a value of _blank, call the method to render as external link
		isExternalLink ? renderAsExternalLink() : renderAsLink()

	);

};

export default Link;