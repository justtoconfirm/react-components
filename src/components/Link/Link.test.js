import React from 'react';
import Link from './index';
import renderer from 'react-test-renderer';

it('Link renders correctly', () => {
	
	const tree = renderer.create(

		<Link href="https://www.bbc.co.uk">BBC</Link>

	).toJSON();

	// Create snapshot or generate snapshot if not already created
	expect(tree).toMatchSnapshot();

});