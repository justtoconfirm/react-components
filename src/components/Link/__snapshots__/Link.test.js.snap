// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<a
  className="link"
  href="https://www.bbc.co.uk"
>
  BBC
</a>
`;
